### DISCUSSION TITLE
how to do abstract event properly

### DISCUSSION THREAD

**Discussion by deltanedas** [space-wizards/space-station-14#16305] [Technical]
i have an abstract event StingEvent which ExtractionStingEvent extends
i try to raise the ExtractionStingEvent but cant handle it, i can only handle the abstract StingEvent

also both events are ByRefEvent if it matters

**Comment by ElectroJr** [comment]
I think the event bus doesn't support subscribing to abstract/non-sealed events. There should probably be some analyser that checks for subscriptions like that and errors, unless I'm wrong or this will change.

You can raise abstract/base events, but its somewhat counterintuitive because you need to cast the event to an `object` first. You can see in the code snippet below that object events are handled slightly differently. It also looks like this might not work for by-ref events in general, but as its not a struct you could just not raise by ref anyways. 

![rider64_uvDgmkj2Go](https://github.com/space-wizards/space-station-14/assets/60421075/5ffbba37-2fbd-48ca-a36b-f6952d37fbca)
